substrate/frame/revive/src/exec.rs

```diff
@@ -2030,13 +2029,8 @@ where
 			tracer.log_event(contract, &topics, &data);
 		});
 
-		if eth_block_storage::is_executing_ethereum_call() {
-			InflightEthTxEvents::<T>::append(EventLog {
-				contract,
-				data: data.clone(),
-				topics: topics.clone(),
-			});
-		}
+		// Capture the log only if it is generated by an Ethereum transaction.
+		eth_block_storage::capture_ethereum_log(contract, &data, &topics);
```

Comment:
The events are RLP encoded into memory to keep only the relevant info around. This optimizes the memory a bit (since we no longer need the full vector of raw logs + the encoded raw logs), but we could still have unbounded growth here. Is there a way to:
- determine a reasonable number of events we allow per transaction without losing compatibility?
- halt / interrupt the evm transaction execution and report the transaction as failed if it goes beyond that threshold? 

Would love to get your thoughts on this 🙏 @pgherveou @athei